src/PantheonTerminalWindow.vala

```diff
@@ -676,14 +678,24 @@ namespace PantheonTerminal {
 
             PantheonTerminal.saved_state.tabs = {};
 
+            int focus = PantheonTerminal.saved_state.focused_tab.clamp(0, tabs.length - 1);
             Idle.add_full (GLib.Priority.LOW, () => {
+                focus += notebook.n_tabs;
```

Comment:
I am not sure about this.  Normally if `notebook.n_tabs` is greater than 0 then terminal is being launched with a specified working directory in which case focus_restored_tabs is false and `focus` is not used anyway.  Is there a case when this not so?  Not that it does any harm.

Reply:
No there is no case when this happens, I have included this so if the behaviour changes it will still work.